Firestore d3 doughnut chart

IsaackMwenda opened the pull request.

ritazagoni commented on Budget Planner/Readme.md:
```diff
@@ -0,0 +1,17 @@
+- - Event listeners
+- - Mouse Over & Click Events
+- - Tooltips
+
```
Could you change the Readme to describe features and how to use?

ritazagoni commented on Budget Planner/graph.js:
```diff
@@ -0,0 +1,178 @@
+const dims={height:300,width:300,radius:150};
+const cent ={x:(dims.width/2 + 5), y:(dims.height / 2 + 5)};
+
```
Please make spacing consistent (here around = and { signs), throughout the files.

ritazagoni commented on Budget Planner/graph.js:
```diff
@@ -0,0 +1,178 @@
+    .data(pie(data));
+
+    //console.log(paths);
+
```
Pls remove debug code

ritazagoni commented on Budget Planner/graph.js:
```diff
@@ -0,0 +1,178 @@
+
+graph.call(tip);
+
+//Update function
```
Would be useful to describe instead what the function is achieving

ritazagoni commented on Budget Planner/graph.js:
```diff
@@ -0,0 +1,178 @@
+//data array and firestore
+var data =[];
+
+budgetdb.collection('projectbudgetplanner').orderBy('cost').onSnapshot(res =>{
```
No need for double newline

ritazagoni commented on Budget Planner/graph.js:
```diff
@@ -0,0 +1,178 @@
+        }
+    });
+
+    // call the update function
```
As above

ritazagoni commented on Budget Planner/graph.js:
```diff
@@ -0,0 +1,178 @@
+
+// use function keyword to allow use of 'this'
+function arcTweenUpdate(d){
+    //console.log(this._current,d);
```
Pls remove debug code

ritazagoni commented on Budget Planner/index.html:
```diff
@@ -0,0 +1,80 @@
+  <script src="index.js"></script>
+  <script src="graph.js"></script>
+</body>
+</html>
```
Newline needed at the end of file

ritazagoni commented on Budget Planner/index.js:
```diff
@@ -0,0 +1,26 @@
+  } else {
+    error.textContent = 'Please enter values before submitting';
+  }
+});
```
Newline needed

ritazagoni commented on Budget Planner/graph.js:
```diff
@@ -0,0 +1,178 @@
+    // i(t) returns a value of d (data object) which we pass to arcPath
+        return arcPath(i(t));
+    };
+     
```
Whitespace can be removed

ritazagoni commented on Budget Planner/graph.js:
```diff
@@ -0,0 +1,178 @@
+    d3.select(n[i])
+        .transition('changeSlicefill').duration(300)
+            .attr('fill','#ffff');
+
```
Whitespace can be removed

ritazagoni commented on Budget Planner/graph.js:
```diff
@@ -0,0 +1,178 @@
+    d3.select(n[i])
+        .transition('changeSlicefill').duration(300)
+            .attr('fill',colour(d.data.name));
+
```
Whitespace can be removed

IsaackMwenda commented:
@ritazagoni have removed whitespaces, added a newline at the end of each file, code comments
and updated readme with more explanation on how to use the tool.

lukechurch commented:
(I'll close it now to clear up the review queue - if you want we can always reopen it)
